InteractionBase/src/commands/CAddNodeToViewByName.cpp

```diff
@@ -40,18 +41,42 @@ bool CAddNodeToViewByName::canInterpret(Visualization::Item*, Visualization::Ite
 	return commandTokens.size() >= 1 && commandTokens.first() == name();
 }
 
-CommandResult* CAddNodeToViewByName::execute(Visualization::Item*, Visualization::Item* target,
-		const QStringList& commandTokens, const std::unique_ptr<Visualization::Cursor>&)
+CommandResult* CAddNodeToViewByName::execute(Visualization::Item* source, Visualization::Item* target,
+		const QStringList& commandTokens, const std::unique_ptr<Visualization::Cursor>& cursor)
 {
 	if (commandTokens.size() < 2)
 		return new CommandResult(new CommandError("Please specify a node to add"));
 
+	auto currentView = target->scene()->currentViewItem();
+	QPoint posToInsert;
+	if (source == currentView && isHorizontal(cursor->region()))
+	{
+		auto below =  findItemBelow(currentView, cursor->region());
```

Comment:
Oh, sorry, as an argument you have to always use the currentView (the item that defines the layout form element), regardless of what's selected.

This approach will only work if the focus is on one of the items. If the focus is on the `currentView` itself, then you should use what I suggested before `dynamic_cast<LayoutCursor*>(cursor)->x()/y()/index()`